test/TestAssets/performance/Perfy.TestAdapter/Perfy.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+// This dll is a TestProject as well as a test adapter.We use it as a baseline for performance.
+// It does no actual discovery or run.It just returns as many results as we ask it to by TEST_COUNT env variable.
+// It has almost no overhead, so all the overhead we see is coming from TestPlatform.
+
+using System.Data;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.Adapter;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.Logging;
+
+namespace PerfyPassing
+{
+    [ExtensionUri(Id)]
+    [DefaultExecutorUri(Id)]
+    public class Perfy : ITestDiscoverer, ITestExecutor
+    {
+        static Perfy()
+        {
+            // No meaning to the number, it is just easy to find when it breaks. Better than returning 1000 or 0 which are both
+            // less suspicious. It could throw, but that is bad for interactive debugging.
+            Count = int.TryParse(Environment.GetEnvironmentVariable("TEST_COUNT") ?? "356", out var count) ? count : 356;
```

Comment:
nit: Since we have no preference for the number, can we still use something like a power of 2, or maybe a prime number ?

Reply:
Also, can we make "TEST_COUNT" a constant ?

Reply:
Why power of 2 or prime number? :)

Why constant and where it would go? I don't want to put it into PerfTestBase, and there are 4 tests that use it in 2 different classes. If you get it wrong, the test will fail. 

In the code above it is used in a single place in single file, so making it a constant only adds mental overhead.

Reply:
I merged it to start measuring. Let me know if you feel strongly about this, and I can make another PR for it.